TKGame/Players/Components/C_Player_WeaponAttack.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework.Input;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using TKGame.BackEnd;
+using TKGame.Components.Interface;
+
+namespace TKGame.Players.Components
+{
+    internal class C_Player_WeaponAttack : IGraphicsComponent
```

Comment:
Why is this a graphics component? If it's because it should be updating the graphics then the logic should be moved to the player graphics in some way.